### PR TITLE
🎨added py.typed to support mypy

### PR DESCRIPTION
**Pull request by sanderegg**
### Issue:
The dockerized mypy is unable to follow on imports, therefore any static typing issue is ignored when for example a service uses wrongly a method defined in a library (e.g. resource-usage-tracker wrong usage of aws-library s3 client).
### Changes
- Adds mypy 1.11.0 as a base package of the tooling in every package,
- Changed the Makefile recipe to use mypy directly inplace instead of through the docker container,
- Removed the mypy docker container as it is useless and dangerous,
- Upgraded the tooling in general
- Added all the py.typed in our base libraries

### Necessary changes in how we define test requirements:
- for mypy to work nicely with our libraries,
- py.typed must be added in any library to indicate that it is statically typed,
- for external libraries, either they provide it or sometimes there are type libraries (such as types-aiofiles, or types-requests), they must be added in the _test.in file so that mypy can use these types.

### Follow-up:
- https://github.com/ITISFoundation/osparc-simcore/issues/6115

###  Highlights on updated libraries (only updated libraries are included)

- #packages before ~ 4
- #packages after ~ 5

|#|name|before|after|upgrade|count|packages|
|-|----|------|-----|-------|-----|--------|
|   1 | astroid                   | 3.2.3           | 3.2.4      |            | 31 | agent🔧</br>api-server🔧</br>autoscaling🔧</br>aws-library🔧</br>catalog🔧</br>clusters-keeper🔧</br>dask-sidecar🔧</br>dask-task-models-library🔧</br>datcore-adapter🔧</br>director-v2🔧</br>dynamic-scheduler🔧</br>dynamic-sidecar🔧</br>e2e-playwright🧪</br>efs-guardian🔧</br>invitations🔧</br>migration🔧</br>models-library🔧</br>notifications-library🔧</br>osparc-gateway-server🔧🧪</br>payments🔧</br>postgres-database🔧</br>public-api🧪</br>resource-usage-tracker🔧</br>service-integration🔧</br>service-library🔧</br>settings-library🔧</br>simcore-sdk🔧</br>storage🔧</br>swarm-deploy🧪</br>web🔧 |
|   2 | pylint                    | 3.2.5           | 3.2.6      |            | 31 | agent🔧</br>api-server🔧</br>autoscaling🔧</br>aws-library🔧</br>catalog🔧</br>clusters-keeper🔧</br>dask-sidecar🔧</br>dask-task-models-library🔧</br>datcore-adapter🔧</br>director-v2🔧</br>dynamic-scheduler🔧</br>dynamic-sidecar🔧</br>e2e-playwright🧪</br>efs-guardian🔧</br>invitations🔧</br>migration🔧</br>models-library🔧</br>notifications-library🔧</br>osparc-gateway-server🔧🧪</br>payments🔧</br>postgres-database🔧</br>public-api🧪</br>resource-usage-tracker🔧</br>service-integration🔧</br>service-library🔧</br>settings-library🔧</br>simcore-sdk🔧</br>storage🔧</br>swarm-deploy🧪</br>web🔧 |
|   3 | ruff                      | 0.5.2           | 0.5.5      |            | 31 | agent🔧</br>api-server🔧</br>autoscaling🔧</br>aws-library🔧</br>catalog🔧</br>clusters-keeper🔧</br>dask-sidecar🔧</br>dask-task-models-library🔧</br>datcore-adapter🔧</br>director-v2🔧</br>dynamic-scheduler🔧</br>dynamic-sidecar🔧</br>e2e-playwright🧪</br>efs-guardian🔧</br>invitations🔧</br>migration🔧</br>models-library🔧</br>notifications-library🔧</br>osparc-gateway-server🔧🧪</br>payments🔧</br>postgres-database🔧</br>public-api🧪</br>resource-usage-tracker🔧</br>service-integration🔧</br>service-library🔧</br>settings-library🔧</br>simcore-sdk🔧</br>storage🔧</br>swarm-deploy🧪</br>web🔧 |
|   4 | setuptools                | 71.0.2          | 71.1.0     | *minor*    | 20 | catalog🔧</br>dask-task-models-library🔧</br>datcore-adapter🔧</br>director-v2🔧</br>dynamic-scheduler🔧</br>dynamic-sidecar🔧</br>e2e-playwright🧪</br>invitations🔧</br>migration🔧</br>models-library🔧</br>notifications-library🔧</br>osparc-gateway-server🔧🧪</br>payments🔧</br>postgres-database🔧</br>public-api🧪</br>service-integration🔧</br>service-library🔧</br>settings-library🔧</br>swarm-deploy🧪 |

*Legend*: 
- ⬆️ base dependency (only services because packages are floating)
- 🧪 test dependency
- 🔧 tool dependency

## Related issue/s
- fixes https://github.com/ITISFoundation/osparc-simcore/issues/6113

## Dev-ops checklist

- [ ] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: .github/workflows/ci-testing-deploy.yml [hunk 1]
```diff
@@ -894,6 +899,7 @@ jobs:
         run: ./ci/github/unit-testing/aws-library.bash install
       - name: typecheck
         run: ./ci/github/unit-testing/aws-library.bash typecheck
```
Comment (matusdrobuliak66) [inline .github/workflows/ci-testing-deploy.yml:901]: Hmm I can see that some of the new services such as resource-usage-tracker, efs-guardian, cluster-keeper are missing here, I guess we should add them here?

**Review by matusdrobuliak66 — APPROVED** [review]

File: .github/workflows/ci-testing-deploy.yml (hunk 1 above)
Comment (sanderegg) [reply .github/workflows/ci-testing-deploy.yml:901]: I am not a big fan of these bash scripts. I think they are a lot of duplicated code. Probably we should think if it is not possible to have only one script with arguments. Also these scripts usually tend to be slightly different than the commands we execute locally.

**Review by GitHK — APPROVED** [review]
Thanks for taking care of this!

File: packages/models-library/src/models_library/service_settings_labels.py [hunk 2]
```diff
@@ -329,7 +329,7 @@ class DynamicSidecarServiceLabels(BaseModel):
     )
 
     callbacks_mapping: Json[CallbacksMapping] | None = Field(
-        default_factory=CallbacksMapping,
+        default_factory=CallbacksMapping,  # type: ignore[arg-type] # this one ANE I am not sure about
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply packages/models-library/src/models_library/service_settings_labels.py:332]: Well the problem here is the CallbacksMapping. It is a type, not a callable. if you are sure this works as intended you can keep the ignore, but this one was worrisome

File: packages/models-library/src/models_library/api_schemas_directorv2/clusters.py [hunk 3]
```diff
@@ -112,7 +112,7 @@ class ClusterDetailsGet(ClusterDetails):
 
 
 class ClusterCreate(BaseCluster):
-    owner: GroupID | None
+    owner: GroupID | None  # type: ignore[assignment]
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply packages/models-library/src/models_library/api_schemas_directorv2/clusters.py:115]: The problem here is that mypy does not like the fact that we override the base field

File: services/api-server/src/simcore_service_api_server/models/domain/projects.py [hunk 4]
```diff
@@ -17,3 +10,5 @@
     "SimCoreFileLink",
     "StudyUI",
 )
+
+# nopycln: file
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/api-server/src/simcore_service_api_server/models/domain/projects.py:14]: thanks for catching that one. I forgot to fix it the right way.

**Pull request merged**